src/dict.c

```diff
@@ -81,6 +83,18 @@ static inline keyCmpFunc dictGetKeyCmpFunc(dict *d) {
     return dictDefaultCompare;
 }
 
+static inline keyCmpFuncWithLen dictGetKeyCmpFuncWithLen(dict *d) {
+    if (d->type->keyCompareWithLen)
+        return d->type->keyCompareWithLen;
+    return NULL;
```

Comment:
```suggestion
    return d->type->keyCompareWithLen;
```

same as dictGetKeyLenFunc(), on the other hand, maybe we don't need these two methods, which were just used in one place.